[yongkotaro] iP

yongkotaro opened the pull request:
# DukePro

*"Your mind is for having ideas, not holding them."* – David Allen (source)

DukePro frees your mind from having to remember things you need to do. It's:

- Text-based
- Easy to learn
- ~~FAST~~ **_SUPER_ FAST** to use

All you need to do is:

1. [Download it from here.](#) <!-- Update link -->
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you 😉
   
And it is **FREE**!

## Features:

- [x] Managing tasks
- [x] Managing deadlines 
- [x] Managing events 
- [ ] Reminders *(coming soon)*

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

yongkotaro committed:
Allow users to mark and unmark tasks.

yongkotaro committed:
Allow users to mark and unmark tasks.

yongkotaro committed:
Allow users to define what type of task they want.

yongkotaro committed:
Allow users to define the tasks they want.

yongkotaro committed:
Handling errors for user misinputs.

yongkotaro committed:
Can delete tasks from the list.

yongkotaro committed:
Edited the EXPECTED.TXT file

yongkotaro committed:
User input is saved in the data/duke.txt file

yongkotaro committed:
User input is saved in data/duke.txt file

yongkotaro committed:
Merge pull request #1 from yongkotaro/branch-Level-7

Branch level 7

yongkotaro committed:
Level 8. Dates and Times
Take in user input for dates in "yyyy-mm-dd" format and converts it to
"MMM dd yyyy" format.

yongkotaro committed:
Merge pull request #2 from yongkotaro/branch-Level-8

Level 8. Dates and Times

maze508 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,49 @@
+import java.time.LocalDate;
```
Javadocs not present in some classes and methods. While there are some comments for methods and classes, the format should look something like this:

```java
/**
 * Calculates the area of a rectangle.
 * 
 * @param width The width of the rectangle. Must be greater than zero.
 * @param height The height of the rectangle. Must be greater than zero.
 * @return The area of the rectangle calculated using the formula {@code width * height}.
 * @throws IllegalArgumentException if either {@code width} or {@code height} is less than or equal to zero.
 */
public double calculateRectangleArea(double width, double height) {
    if (width <= 0 || height <= 0) {
        throw new IllegalArgumentException("Width and height must be greater than zero.");
    }
    return width * height;
}

```

maze508 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,14 @@
+    }
+
+    public String toString() {
+        return "[E]" + "[" + getStatusIcon() + "] " + this.description + " (from: " + stringToDate(this.start) + " to: " +
```
Line length should be no longer than 120 charecters. You can consider using linebreak

maze508 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,225 @@
+import java.io.*;
```
Imported classes should always be listed explicitly, avoid using `import java.io.*`

maze508 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,225 @@
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.Iterator;
```
Good consistent ordering of import statements!

maze508 reviewed (commented):
You have somewhat closely followed the Java Coding Standard, keep up the good work!

yongkotaro committed:
Duke is more object-orientated.

yongkotaro committed:
Merge branch 'branch-A-MoreOOP'

yongkotaro committed:
Put the classes in duke package.

yongkotaro committed:
Edited duke.txt file.

yongkotaro committed:
Merge remote-tracking branch 'origin/add-gradle-support'

yongkotaro committed:
Used Gradle to build Duke.

yongkotaro committed:
Merge pull request #3 from yongkotaro/branch-A-Packages

Branch a packages

yongkotaro committed:
Merge branch 'master' of https://github.com/yongkotaro/ip

yongkotaro committed:
Added JUnit tests.

yongkotaro committed:
Added JUnit tests.

yongkotaro committed:
Created JAR file for duke.

yongkotaro committed:
Added JavaDocs to classes and their methods.

yongkotaro committed:
Reformatted switch case to follow Java coding
standard.

yongkotaro committed:
Allow users to find a task by searching for a
keyword using "find 'keyword'".

yongkotaro committed:
Merge branch 'branch-A-JavaDoc'

yongkotaro committed:
Merge branch 'branch-A-CodingStandard'

wongkj12 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,151 @@
+
+import duke.Event;
+
+import java.io.File;
```
Perhaps java.io imports could be grouped together for clarity

wongkj12 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,24 @@
+     * @return the string output of the Event task
+     */
+    public String toString() {
+        return "[E]" + "[" + getStatusIcon() + "] " + this.description + " (from: " + stringToDate(this.start) + " to: " +
```
Line length exceeds 110 chars, perhaps the line could be wrapped a bit earlier

wongkj12 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,74 @@
+ */
+public class Task {
+    protected String description;
+    protected int isDone;
```
Might be better for isDone to be a boolean (if values other than 0 or 1 are not used)

wongkj12 reviewed (commented):
Aside from a few minor code quality issues, your code looks good and is easy to read.

Wongqingyee commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,62 @@
+ * Ui class handles the user interface of the application.
+ */
+public class Ui {
+    protected Scanner in;
```
Perhaps can give a name like userInput, for example.
A name is not just for differentiation; it should explain the named entity to the reader accurately and at a sufficient level of detail. (can refer Code Quality: Naming)

Wongqingyee commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,151 @@
+     * @return An ArrayList of tasks.
+     */
+    public ArrayList<Task> load() {
+        ArrayList<Task> list = new ArrayList<>();
```
The name can be lists instead of list. 
Distinguish clearly between single-valued and multi-valued variables. (can refer Code Quality: Namings)

Wongqingyee commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,151 @@
+     * @throws IOException If an I/O error occurs.
+     */
+    public void addToFile(String textToAdd) throws IOException {
+        BufferedWriter fw = new BufferedWriter(new FileWriter(this.path, true));
```
Can have a more explicit name.
It should explain the named entity to the reader accurately and at a sufficient level of detail (can refer Code Quality: Namings).

Wongqingyee commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,144 @@
+        } else {
+            tasks.add(new Event(description, 0, start, end));
+            String task = tasks.get(tasks.size() - 1).toString();
+            String numberOfTasks = "Now you have "
```
Can have a name like taskMessage, since it is a String.

yongkotaro committed:
Refactored code.

yongkotaro committed:
Refactored code.

yongkotaro committed:
Refactored code.

yongkotaro committed:
Added dependencies for JavaFX in gradle.

yongkotaro committed:
Implemented JavaFX GUI for Duke.

yongkotaro committed:
Implemented JavaFX GUI for Duke.

yongkotaro committed:
Updated FXML files.

yongkotaro committed:
Implemented GUI for Duke.

yongkotaro committed:
Refactored code

Created specific Command subclasses for the
different types of commands.

Added JavaDocs for the remaining methods.

yongkotaro committed:
Remove exception handling in TaskList functions

Shifted exception handling to the Command subclasses.

yongkotaro committed:
Merge pull request #5 from yongkotaro/branch-A-CodingQuality

Remove exception handling in TaskList functions

yongkotaro committed:
Add assert statements

Create assert statements for adding and deleting tasks to check if
the code works as intended.

yongkotaro committed:
Resolve merge conflict

Resolve merge conflict in deleteTask method
between master and A-Assertions branch.

yongkotaro committed:
Merge pull request #6 from yongkotaro/branch-A-Assertions

Add assert statements

yongkotaro committed:
Resolve error in TaskList class

Fix error in deleteTask method in the TaskList
class.

yongkotaro committed:
Add C-Help extension

Users can type "help" to get a list of available
commands.

yongkotaro committed:
Merge pull request #7 from yongkotaro/branch-BCD

Add C-Help extension

yongkotaro committed:
Make getter methods for Task class

Make the getter methods for description and isDone
value for tasks for JUnit tests.

yongkotaro committed:
Add JavaDocs for methods

Add JavaDocs for methods in the Duke, MainWindow
and Storage class

yongkotaro committed:
Edit error message

Edit error message in the case where there was an
issue in creating the file in the Storage class.

yongkotaro committed:
Update README

Update README folder to include the commands the
user can use.

yongkotaro committed:
Rename Ui.png file

Rename Ui.png.png to Ui.png.

yongkotaro committed:
Add duke.txt file to gitignore

Add duke.txt file to gitignore.

yongkotaro committed:
Add JavaDocs for TaskList class

Add JavaDocs for methods in TaskList class.

yongkotaro committed:
Rename exception class

Rename DukeException to TaroException.

yongkotaro committed:
Edit exception handling in Storage class

Edit exception handling in Storage class.
Give credit for referred code.

yongkotaro committed:
Merge pull request #8 from yongkotaro/branch-A-CodeQuality

Edit exception handling in Storage class

yongkotaro committed:
Add assert statement for TaskList methods

Add assert statement for markTask and unmarkTask
method to check if the corresponding task is
marked or unmarked respectively.

yongkotaro committed:
Merge pull request #9 from yongkotaro/branch-A-Assertions

Add assert statement for TaskList methods

yongkotaro committed:
Update README.md

Edit file path for screenshot of the GUI.

yongkotaro committed:
Update README.md
